lib/resourcebuilder/resourcebuilder.go

```diff
@@ -130,10 +132,11 @@ func (b *builder) Do(ctx context.Context) error {
 			if err := b.modifyDeployment(ctx, typedObject); err != nil {
 				return err
 			}
-			if _, _, err := resourceapply.ApplyDeploymentv1(ctx, b.appsClientv1, typedObject, reconcilingMode); err != nil {
+			if actual, _, err := resourceapply.ApplyDeploymentv1(ctx, b.appsClientv1, typedObject, reconcilingMode); err != nil {
 				return err
+			} else if actual != nil {
```

Comment:
So before your change if a resource was `IsCreateOnly` this would have segfault'ed? Seems like a pretty big hole.

Should the `nil` check be in each of the functions instead?

Reply:
ACK. Did not notice that param change.